react/layout.jsx

```diff
@@ -45,13 +54,15 @@ let Layout = React.createClass({
         yield dattcore.asyncInitialize()
 
         let userSetupFlag = yield dattcore.asyncGetUserSetupFlag()
-        if (!userSetupFlag && (window.location.hash === '#/frontpage' || !window.location.hash || window.location.hash === '#/')) {
+
+        if (!userSetupFlag && !this.state.skipSetup && (window.location.hash === '#/frontpage' || !window.location.hash || window.location.hash === '#/') && (window.location.toString().indexOf('tests.html') === -1)) {
```

Comment:
Another thought occurs to me about "window" - this means this file can't be easily tested in node.js either. It's very convenient for files to be able to be tested in node.js since the tests run faster that way. The UI should really be tested as thoroughly as dattcore - the only reason why it wasn't it because I didn't know a good way to test it, since I am inexperienced with React.